core/src/test/scala/unit/kafka/server/OffsetFetchRequestTest.scala

```diff
@@ -14,233 +14,532 @@
+
       }
     }
   }
 
-  @Test
-  def testOffsetFetchRequestWithMultipleGroups(): Unit = {
-    createTopic(topics(0))
-    createTopic(topics(1), numPartitions = 2)
-    createTopic(topics(2), numPartitions = 3)
-
-    // create 5 consumers to commit offsets so we can fetch them later
-    val partitionMap = groupToPartitionMap.asScala.map(e => (e._1, Option(e._2).getOrElse(allTopicsList)))
-    groups.foreach { groupId =>
-      consumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId)
-      commitOffsets(partitionMap(groupId))
+  private def testMultipleGroupsOffsetFetch(useNewProtocol: Boolean, requireStable: Boolean): Unit = {
+    if (useNewProtocol && !isNewGroupCoordinatorEnabled) {
+      fail("Cannot use the new protocol with the old group coordinator.")
+    }
+
+    val admin = cluster.createAdminClient()
+
+    // Creates the __consumer_offsets topics because it won't be created automatically
+    // in this test because it does not use FindCoordinator API.
+    TestUtils.createOffsetsTopicWithAdmin(
+      admin = admin,
+      brokers = if (cluster.isKRaftTest) {
+        cluster.asInstanceOf[RaftClusterInstance].brokers.collect(Collectors.toList[KafkaBroker]).asScala
+      } else {
+        cluster.asInstanceOf[ZkClusterInstance].servers.collect(Collectors.toList[KafkaBroker]).asScala
+      }
+    )
+
+    // Create the topic.
+    TestUtils.createTopicWithAdminRaw(
+      admin = admin,
+      topic = "foo",
+      numPartitions = 3
+    )
+
+    // Create groups and commits offsets.
+    List("grp-0", "grp-1", "grp-2").foreach { groupId =>
+      val (memberId, memberEpoch) = if (useNewProtocol) {
+        joinConsumerGroupWithNewProtocol(groupId)
+      } else {
+        joinConsumerGroupWithOldProtocol(groupId)
+      }
+
+      for (partitionId <- 0 to 2) {
+        commitOffset(
+          groupId = groupId,
+          memberId = memberId,
+          memberEpoch = memberEpoch,
+          topic = "foo",
+          partition = partitionId,
+          offset = 100L + partitionId,
+          expectedError = Errors.NONE,
+          version = ApiKeys.OFFSET_COMMIT.latestVersion(isUnstableApiEnabled)
+        )
+      }
     }
 
-    for (version <- 8 to ApiKeys.OFFSET_FETCH.latestVersion()) {
-      val request =  new OffsetFetchRequest.Builder(groupToPartitionMap, false, false)
-        .build(version.asInstanceOf[Short])
-      val response = connectAndReceive[OffsetFetchResponse](request)
-      response.data.groups.asScala.map(_.groupId).foreach( groupId =>
-        verifyResponse(response.groupLevelError(groupId), response.partitionDataMap(groupId), partitionMap(groupId))
+    // Start from version 8 because older versions do not support
+    // fetch offsets for multiple groups.
+    for (version <- 8 to ApiKeys.OFFSET_FETCH.latestVersion(isUnstableApiEnabled)) {
+      assertEquals(
+        List(
+          // Fetch foo-0, foo-1 and foo-5.
```

Comment:
Since we're already checking offset fetch for foo-0 and foo-1 in the next fetchAllPartitions block, did we wanna check for a mix of existing and non existent partitions?, can we also comment fetch non-existent partition foo-5